optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/drools/DroolsScoreDirector.java

```diff
@@ -190,12 +148,24 @@ public boolean isConstraintMatchEnabled() {
         return workingScoreHolder.getConstraintMatchTotals();
     }
 
+    @Override
+    public Map<String, ConstraintMatchTotal> getConstraintMatchTotalMap() {
+        if (workingSolution == null) {
+            throw new IllegalStateException(
+                    "The method setWorkingSolution() must be called before the method getConstraintMatchTotalMap().");
+        }
+        // Notice that we don't trigger the variable listeners
+        kieSession.fireAllRules();
+        return workingScoreHolder.getConstraintMatchTotalMap();
+    }
+
     @Override
     public Map<Object, Indictment> getIndictmentMap() {
         if (workingSolution == null) {
             throw new IllegalStateException(
                     "The method setWorkingSolution() must be called before the method getIndictmentMap().");
         }
+        // Notice that we don't trigger the variable listeners
```

Comment:
Noticed. Perhaps you could also explain why it is important to notice that?

Reply:
Yes, I agree. But only my former self currently knows why... (which really proves your point).
This is not an issue introduced in this PR.
Deferred into https://issues.jboss.org/browse/PLANNER-1584